src/views/tutorial-view/index.tsx

```diff
@@ -33,64 +28,81 @@ import TutorialsSidebar, {
 import TabProvider from 'components/tabs/provider'
 import TutorialMeta from 'components/tutorial-meta'
 import VideoEmbed from 'components/video-embed'
-import InstruqtProvider from 'contexts/instruqt-lab'
-import { useOptInAnalyticsTracking } from 'hooks/use-opt-in-analytics-tracking'
 import { getLearnRedirectPath } from 'components/opt-in-out/helpers/get-learn-redirect-path'
 
 // Local imports
+import {
+	CollectionContext,
+	LayoutContentWrapperProps,
+	TutorialData,
+	TutorialSidebarSidecarProps,
+	TutorialViewProps,
+} from './types'
 import MDX_COMPONENTS from './utils/mdx-components'
 import { formatTutorialToMenuItem, generateCanonicalUrl } from './utils'
+import getVideoUrl from './utils/get-video-url'
+import { getCanonicalCollectionSlug } from './utils/get-canonical-collection-slug'
 import {
 	FeaturedInCollections,
 	NextPrevious,
 	getNextPrevious,
 } from './components'
-import getVideoUrl from './utils/get-video-url'
-import { getCanonicalCollectionSlug } from './utils/get-canonical-collection-slug'
 import s from './tutorial-view.module.css'
 
-export interface TutorialViewProps {
-	layoutProps: TutorialSidebarSidecarProps
-	product: LearnProductData
-	tutorial: TutorialData
-}
+/**
+ * The purpose of this wrapper component is to make it possible to invoke the
+ * `useMobileMenu` hook. It cannot be invoked in `TutorialView`. This is because
+ * it requires being invoked within a `MobileMenuProvider`.
+ *
+ * `MobileMenuProvider` is rendered by `CoreDevDotLayout`, which is indirectly
+ * rendered by `SidebarSidecarLayout`. This means the `useMobileMenu` hook can
+ * only be invoked here by a component rendered within `SidebarSidecarLayout`.
+ */
+const LayoutContentWrapper = ({
+	children,
+	collectionCtx,
+	product,
+	setCollectionViewSidebarSections,
+}: LayoutContentWrapperProps) => {
+	const { mobileMenuIsOpen } = useMobileMenu()
+	const haveLoadedData = useRef(false)
 
-export interface TutorialData
-	extends Pick<
-		ClientTutorial,
-		| 'name'
-		| 'slug'
-		| 'readTime'
-		| 'productsUsed'
-		| 'edition'
-		| 'handsOnLab'
-		| 'video'
-	> {
-	collectionCtx: CollectionContext
-	content: MDXRemoteSerializeResult
-	nextCollectionInSidebar?: ClientCollectionLite
-}
+	/**
+	 * Only need to load the data once, on the first open of the mobile menu
+	 */
+	if (haveLoadedData.current === false && mobileMenuIsOpen) {
+		getCollectionViewSidebarSections(product, collectionCtx.current).then(
+			(result: CollectionCategorySidebarSection[]) => {
+				haveLoadedData.current = true
+				setCollectionViewSidebarSections(result)
+			}
+		)
+	}
```

Comment:
💭 This should be in a `useEffect` instead of the component body to avoid a side effect during render. We should also add a `.catch` here to ensure we handle a scenario where the fetch fails.

Reply:
@BRKalow I added `useEffect` and a `catch` in https://github.com/hashicorp/dev-portal/pull/709/commits/642bcf2ca6699944af431f7db9299c17e066931f. What do we want to do if there is an error? If we do nothing, the sidebar where this data shows will not be useful because it'll be just about empty. So I think we _want_ the error to happen.

<details>
  <summary>Screenshot of empty sidebar</summary>
  <img width="411" src="https://user-images.githubusercontent.com/43934258/179268658-abffdca7-6ecd-46d4-8aec-cb55af4a6327.png">
</details>

Since this is a UX discussion we probably want to have more widely, can we add a `TODO` in the `catch` block with an Asana task and come back so this PR isn't blocked?

Reply:
The intent with catching the error here is that we control the error state and don't present a totally unexpected UX to the user. I think in the rare case that the call fails, the empty menu is reasonable and better than the page being broken in some other, unexpected way.

Reply:
TODO added in https://github.com/hashicorp/dev-portal/pull/709/commits/68c2fccc8a46c2d42a6e711ea7c50d151fce36f9!